app/services/forest_liana/resources_getter.rb

```diff
@@ -19,6 +19,49 @@ def initialize(resource, params)
       prepare_query()
     end
 
+    def self.get_ids_from_request(params)
+      attributes = params.dig('data', 'attributes')
+      has_body_attributes = attributes != nil
+      is_select_all_records_query = has_body_attributes && attributes[:all_records] === true
+    
+    
+      # NOTICE: If it is not a "select all records" query and it receives a list of ID, return list of ID.
+      return attributes[:ids] if (!is_select_all_records_query && attributes[:ids])
+
+      # NOTICE: If it is a "select all records" we have to perform query to build ID list.
+      ids = Array.new
+
+      # NOTICE: Initialize actual resources getter (could either a HasManyGetter or a ResourcesGetter).
+      isRelatedData = attributes[:parent_collection_id] &&
+        attributes[:parent_collection_name] &&
+        attributes[:parent_association_name]
+      if (isRelatedData)
+        parent_collection_name = attributes[:parent_collection_name]
+        parent_model = ForestLiana::SchemaUtils.find_model_from_collection_name(parent_collection_name)
+        model = parent_model.reflect_on_association(attributes[:parent_association_name].try(:to_sym))
+        resources_getter = ForestLiana::HasManyGetter.new(parent_model, model, attributes.merge({ 
+          id: attributes[:parent_collection_id], 
+          association_name: attributes[:parent_association_name], 
+          collection: parent_collection_name
+        }))
+      else
+        collection_name = attributes[:collection_name]
+        model = ForestLiana::SchemaUtils.find_model_from_collection_name(collection_name)
+        resources_getter = ForestLiana::ResourcesGetter.new(model, attributes)
+      end
+
+      # NOTICE: build ID list.
+      resources_getter.query_for_batch.find_in_batches() do |records|
+        ids += records.map { |record| record.id }
+      end
+
+      # NOTICE: remove excluded IDs.
+      ids_excluded = (attributes[:all_records_ids_excluded]).map { |id_excluded| id_excluded.to_s }
```

Comment:
[Question] Is this line really necessary?
I am just wondering if we can simplify this code.

Reply:
Good catch! 

I used the same logic in forest-express: force `ids_exluded` and `ids` to strings before comparing them (to avoid unexpected behavior with mongo's `ObjectId`s, `int`, `uuid`, etc.). **It's not necessary**(?), but testing and verifying various cases was not that simple and I'm not confident about being aware of all cases. It's overprotective, but I prefer use this to be consistant with `forest-express`, not trusting data that comes from frontend and avoid unexpected deletion. Furthermore, I don't think it's too expensive. So I guess it's OK, still your question was legit!